hydra_python_core/doc_writer.py

```diff
@@ -629,6 +632,46 @@ def generate(self) -> Dict[str, Any]:
         return error
 
 
+class HydraLink():
+    """Template for a link property."""
+
+    def __init__(
+            self, id_: str, title: str = "",
+            desc: str = "", domain: str = "", range_: str = "") -> None:
+        """Initialize the Hydra_Link."""
+        self.id_ = id_ if "http" in id_ else "vocab:{}".format(id_)
+        self.range = range_
+        self.title = title
+        self.desc = desc
+        self.domain = domain
+        self.supportedOperation = list()  # type: List
+
+    def add_supported_op(
+            self, op: Union['EntryPointOp', 'HydraClassOp']) -> None:
+        """Add a new supportedOperation.
+
+        Raises:
+            TypeError: If `op` is not an instance of `HydraClassOp` or `EntryPointOp`
+
+        """
+        if not isinstance(op, (HydraClassOp, EntryPointOp)):
+            raise TypeError("Type is not <HydraClassOp>")
+        self.supportedOperation.append(op)
+
+    def generate(self) -> Dict[str, Any]:
+        """Get the Hydra link as a python dict."""
+        link = {
+            "@id": self.id_,
+            "@type": "hydra:Link",
+            "title": self.title,
+            "description": self.desc,
+            "range": self.range,
+            "domain": self.domain,
+            "supportedOperation": [x.generate() for x in self.supportedOperation],
+        }
+        return link
+
```

Comment:
We need another function here to create instances of hydra:Link. Something similar to the way it is defined in the spec.
https://www.hydra-cg.com/spec/latest/core/#example-3-defining-properties-representing-hyperlinks-using-hydra-s-link-class